arch/xtensa/src/esp32s3/esp32s3_gpio.c

```diff
@@ -171,24 +244,38 @@ void esp32s3_gpio_matrix_in(uint32_t gpio, uint32_t signal_idx, bool inv)
  * Name: esp32s3_gpio_matrix_out
  *
  * Description:
- *   Set signal output to gpio
- *   NOTE: one signal can output to several gpios
- *   If signal_idx == 0x100, cancel output put to the gpio
+ *   Set signal output to GPIO.
+ *   NOTE: one signal can output to several GPIOs.
+ *
+ * Input Parameters:
+ *   pin           - GPIO pin to be configured.
+ *   signal_idx    - Signal index.
+ *                   - If signal_idx == 0x100, cancel output to the GPIO.
+ *   out_inv       - Flag indicating whether the signal output is inverted.
+ *   oen_inv       - Flag indicating whether the signal output enable is
+ *                   inverted.
+ *
+ * Returned Value:
+ *   None.
  *
  ****************************************************************************/
 
-void esp32s3_gpio_matrix_out(uint32_t gpio, uint32_t signal_idx,
-                             bool out_inv, bool oen_inv)
+void esp32s3_gpio_matrix_out(uint32_t pin, uint32_t signal_idx, bool out_inv,
+                             bool oen_inv)
 {
-  uint32_t regaddr = GPIO_FUNC0_OUT_SEL_CFG_REG + (gpio * 4);
+  uint32_t regaddr = GPIO_FUNC0_OUT_SEL_CFG_REG + (pin * 4);
   uint32_t regval = signal_idx << GPIO_FUNC0_OUT_SEL_S;
 
-  if (gpio >= ESP32S3_NGPIOS)
+  DEBUGASSERT(is_valid_gpio(pin));
+
+  if (pin < 32)
     {
-      return;
+      putreg32((UINT32_C(1) << pin), GPIO_ENABLE_W1TS_REG);
+    }
+  else
+    {
+      putreg32((UINT32_C(1) << (pin - 32)), GPIO_ENABLE1_W1TS_REG);
     }
-
-  putreg32(1ul << gpio, GPIO_ENABLE_W1TS_REG);
 
   if (out_inv)
```

Comment:
The `if (signal_idx == 0x100)` check is missing.

Reply:
In fact the hardware register already expects that value and will behave accordingly, so there is no need for performing any validations on the driver level.

Reply:
Then should we add `DEBUGASSERT(signal_idx == 0x100);`? Or that is redundant?

Reply:
I'm just looking into an API description:
```
 *   signal_idx    - Signal index.
 *                   - If signal_idx == 0x100, cancel output to the GPIO.
```
so that is the point of my confusion

Reply:
> Then should we add DEBUGASSERT(signal_idx == 0x100);? Or that is redundant?

The **0x100** may be simply written into the register and the GPIO Matrix peripheral will cancel the output.
So no need for the `DEBUGASSERT` here.
The expected value for `signal_idx` is one of those defined in the `esp32s3_gpio_sigmap.h` header. **0x100** matches `SIG_GPIO_OUT_IDX`.

Reply:
Oah, sorry... I missed the line `uint32_t regval = signal_idx << GPIO_FUNC0_OUT_SEL_S;`. Again, was confused by `esp32s3_gpio_matrix_in` that has a swapped `regaddr` and `regval` calculation. Thank for educating me :)

Reply:
Thank you as well! Sometimes we take some information for granted, so I usually take this opportunity to do further research and learn a bit more hehehe